packages/modules/src/MultichainValidationModule.ts

```diff
@@ -0,0 +1,102 @@
+import { UserOperation, ChainId } from '@biconomy/core-types'
+import { Logger, getUserOpHash } from '@biconomy/common'
+import { ethers } from 'ethers'
+import MerkleTree from 'merkletreejs'
+import { ECDSAOwnershipValidationModule } from './ECDSAOwnershipValidationModule'
+import { MULTICHAIN_VALIDATION_MODULE_ADDRESSES_BY_VERSION } from './utils/Constants'
+import { keccak256, arrayify, defaultAbiCoder, hexConcat, hexZeroPad } from 'ethers/lib/utils'
+import {
+  ECDSAOwnershipValidationModuleConfig,
+  ModuleVersion,
+  MultiChainUserOpDto
+} from './utils/Types'
+export class MultiChainValidationModule extends ECDSAOwnershipValidationModule {
+  // Review: this.chainId should not be in spec of this class
+  // May not inherit from ECDSAOwnershipValidationModule
+
+  constructor(moduleConfig: ECDSAOwnershipValidationModuleConfig) {
+    super(moduleConfig)
+
+    if (moduleConfig.moduleAddress) {
+      this.moduleAddress = moduleConfig.moduleAddress
+    } else if (moduleConfig.version) {
+      const moduleAddr = MULTICHAIN_VALIDATION_MODULE_ADDRESSES_BY_VERSION[moduleConfig.version]
+      if (!moduleAddr) {
+        throw new Error(`Invalid version ${moduleConfig.version}`)
+      }
+      this.moduleAddress = moduleAddr
+      this.version = moduleConfig.version as ModuleVersion
+    }
+  }
+
+  // TODO: Actually depends on the module address so maybe we can make it dynamic
+  getDummySignature(): string {
+    return '0x00000000000000000000000000000000000000000000000000000000000000400000000000000000000000002E817fe3749B81dA801fc08B247E081ec20eB080000000000000000000000000000000000000000000000000000000000000004181d4b4981670cb18f99f0b4a66446df1bf5b204d24cfcb659bf38ba27a4359b5711649ec2423c5e1247245eba2964679b6a1dbb85c992ae40b9b00c6935b02ff1b00000000000000000000000000000000000000000000000000000000000000'
```

Comment:
Why is this needed, what needs to be updated?

Reply:
done